linkerd/app/outbound/src/http/logical/policy/route/backend.rs

```diff
@@ -0,0 +1,102 @@
+use super::{super::Concrete, filters};
+use linkerd_app_core::{proxy::http, svc, Error, Result};
+use linkerd_http_route as http_route;
+use linkerd_proxy_client_policy as policy;
+use std::{fmt::Debug, hash::Hash, sync::Arc};
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub(crate) struct Backend<T, F> {
+    pub(crate) concrete: Concrete<T>,
+    pub(crate) filters: Arc<[F]>,
+}
+
+pub(crate) type Matched<T, M, F> = super::Matched<M, Backend<T, F>>;
+pub(crate) type Http<T> = Matched<T, http_route::http::r#match::RequestMatch, policy::http::Filter>;
+pub(crate) type Grpc<T> = Matched<T, http_route::grpc::r#match::RouteMatch, policy::grpc::Filter>;
+
+// === impl Backend ===
+
+impl<T: Clone, F> Clone for Backend<T, F> {
+    fn clone(&self) -> Self {
+        Self {
+            filters: self.filters.clone(),
+            concrete: self.concrete.clone(),
+        }
+    }
+}
+
+// === impl Matched ===
+
+impl<M, T, F> From<(Backend<T, F>, super::MatchedRoute<T, M, F>)> for Matched<T, M, F> {
+    fn from((params, route): (Backend<T, F>, super::MatchedRoute<T, M, F>)) -> Self {
+        Matched {
+            r#match: route.r#match,
+            params,
+        }
+    }
+}
+
+impl<T, M, F> Matched<T, M, F>
+where
+    // Parent target.
+    T: Debug + Eq + Hash,
+    T: Clone + Send + Sync + 'static,
+    // Request match summary
+    M: Clone + Send + Sync + 'static,
+    // Request filter.
+    F: Clone + Send + Sync + 'static,
+    // Assert that filters can be applied.
+    Self: filters::Apply,
+{
+    pub(crate) fn layer<N, S>() -> impl svc::Layer<
```

Comment:
maybe this should have a comment explaining that this layer will construct an `ApplyFilters` layer by extracting an `Apply` from the target?